lib/temporal/workflow/state_manager.rb

```diff
@@ -94,6 +95,27 @@ def next_event_id
         @last_event_id += 1
       end
 
+      def validate_append_command(command)
+        return if commands.last.nil?
+        _, previous_command = commands.last
```

Comment:
Sorry, I was an idiot; there was no need to loop since we can maintain the list incrementally.